refactor: remove pages and controllers related to pending peer-reviews

lushlife99 opened the pull request:
### Issue Number
<!-- Which issue does this PR address? E.g. "Resolves #123" -->
* Resolves #2082 

### Purpose
<!-- What is the purpose of this PR? Why is it needed? -->
* Remove jsp, controller related to pending peer-reviews

### Technical Details
<!-- Are there any key aspects of the implementation to highlight? -->
* 

### Testing Instructions
<!-- How can the reviewer verify the functionality or fix introduced by this PR? Please provide steps. -->
* 

### Screenshots
<!-- If this PR affects the UI, please include before/after screenshots demonstrating the change(s). -->
* 

<!-- Attribution: External code is properly credited. -->

---

### Format Checks
<!-- If this PR contains files with format violations, run 'mvn spotless:apply' to fix them. -->

> [!NOTE]
> Files in PRs are automatically checked for format violations with `mvn spotless:check`.

If this PR contains files with format violations, run `mvn spotless:apply` to fix them.

jo-elimu commented on src/main/java/ai/elimu/web/content/peer_review/LetterSoundPeerReviewEventCreateController.java:
@lushlife99 The URL in this controller (`/content/letter-sound-peer-review-event/create`) is being used by `\WEB-INF\jsp\content\letter-sound\edit.jsp`, so it shouldn't be deleted.

jo-elimu reviewed (changes requested):
https://github.com/elimu-ai/webapp/pull/2136/files#r2046493005

jo-elimu commented on src/main/java/ai/elimu/web/content/peer_review/Slf4jpublic.java:
@ideal3147 Is this file needed for anything, or can it be deleted?

(ref https://github.com/elimu-ai/webapp/pull/2036)

lushlife99 replied:
Restored a controller and Slf4jpublic.java as they were still being used in other parts of the codebase.
Currently, only the JSP files related to peer-reviews have been removed.

ideal3147 replied:
@jo-elimu 
So sorry for creating unused code...
It can be deleted

jo-elimu commented:
@coderabbitai This PR should not be approved yet, because we need to make some changes first: https://github.com/elimu-ai/webapp/pull/2136#discussion_r2046493005

lushlife99 commented:
Restored a controller and `Slf4jpublic.java` as they were still being used in other parts of the codebase. 
Currently, only the JSP files related to peer-reviews have been removed.

jo-elimu reviewed (changes requested):
@lushlife99 Also remember to remove the links from the UI that are no linker working. I.e. all of those links pointing to `*/peer-reviews`.

lushlife99 commented:
> @lushlife99 Also remember to remove the links from the UI that are no linker working. I.e. all of those links pointing to `*/peer-reviews`.

Changes
Removed all peer-review related controllers

Removed peer-review links from main.jsp

Please let me know if there’s anything else I should update. I'd appreciate your guidance!

jo-elimu reviewed (changes requested):
@lushlife99 Also please remove the "You can also peer-review [...]" links on top of these pages:

* http://eng.elimu.ai/content/letter-sound/list
* http://eng.elimu.ai/content/number/list
* http://eng.elimu.ai/content/word/list

lushlife99 commented:
@jo-elimu 

The last requested change has been applied. Thank you for your review!
